server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -3557,6 +3557,22 @@ public int getReadBufferSizeInBytes() {
         return bufferSize;
     }
 
+    protected static final String REGISTERS_PATH = "registers";
+
+    @Override
+    public long getRegister(String key) throws IOException {
+        final BlobContainer registers = blobStore().blobContainer(basePath().add(REGISTERS_PATH));
+        final byte[] buffer = new byte[8];
+        try (var in = registers.readBlob(key)) {
+            if (Long.BYTES != org.elasticsearch.core.Streams.readFully(in, buffer)) {
+                throw new IllegalStateException("Did not find an 8b long value at key [" + key + "]");
+            }
+        } catch (NoSuchFileException noSuchFileException) {
+            return 0L;
+        }
+        return Numbers.bytesToLong(buffer, 0);
+    }
```

Comment:
It is surprising that we allow reading a register, but not updating it in this implementation

Reply:
It sort of makes sense to me, the write operation implementation will be pretty different across subclasses but the read will always look like this because all the blob stores have atomic writes....

... except for `FsRepository`, I think we need more locking on the read path in this case :)

Reply:
Meh you're right :) I figured it was good enough since an 8b read will always be atomic, but we still have the empty file state to deal with. Let me fix that.

Reply:
Alrighty, used the heavy handed locking from the cas method here now as well.